src/recode.jl

```diff
@@ -76,21 +103,19 @@ function recode!(dest::AbstractArray{T}, src::AbstractArray, default::Any, pairs
             eltype(dest) >: Missing ||
                 throw(MissingException("missing value found, but dest does not support them: " *
                                        "recode them to a supported value"))
-            dest[i] = missing
+            return missing
         elseif default isa Nothing
             try
-                dest[i] = x
+                return convert(T, x)
             catch err
                 isa(err, MethodError) || rethrow(err)
                 throw(ArgumentError("cannot `convert` value $(repr(x)) (of type $(typeof(x))) to type of recoded levels ($T). " *
                                     "This will happen with recode() when not all original levels are recoded " *
                                     "(i.e. some are preserved) and their type is incompatible with that of recoded levels."))
             end
         else
-            dest[i] = default
+            return default
         end
-
-        @label nextitem
```

Comment:
I can't quite see what you're pointing to here, the only time `x` is returned in this implementation as well as in the original is when there are no matches in the recode pairs and `default` is `nothing`. The last `if-elseif-else` block is guaranteed to either return or throw but it would of course be possible to skip the `else` and end with a `return default`.